soccer/planning/tests/AnglePlanningTest.cpp

```diff
@@ -23,24 +32,73 @@ TEST(AnglePlanning, FaceAngle) {
         RobotInstant{Pose(2.5, 0, 0), Twist(1, 0, 0), RJ::Time(2500ms)});
     trajectory.AppendInstant(
         RobotInstant{Pose(3, 0, 0), Twist(1, 0, 0), RJ::Time(3000ms)});
+    return trajectory;
+}
 
+static RotationConstraints make_rotation_constraints() {
     RotationConstraints constraints;
     constraints.maxAccel = 1.0;
     constraints.maxSpeed = 3.0;
+    return constraints;
+}
 
-    auto angle_fn = AngleFns::facePoint(Point(2, 0.5));
-    //    auto angle_fn = AngleFns::faceAngle(1.12);
-    //    auto angle_fn = AngleFns::tangent;
-    PlanAngles(&trajectory, trajectory.first(), angle_fn, constraints);
+// Make a RobotInstant with the given linear motion and exactly coinciding with the correct angle function.
+static RobotInstant make_initial_instant(const LinearMotionInstant& linear,
+                                         const RJ::Time& time,
+                                         const AngleFunction& angle_fn,
+                                         double approx_angle = 0.0) {
+    Eigen::Vector2d gradient;
+    double angle = angle_fn(linear, approx_angle, &gradient);
+    double angular_velocity = gradient.dot(Eigen::Vector2d(linear.velocity));
+    return RobotInstant{Pose{linear.position, angle}, Twist{linear.velocity, angular_velocity}, time};
+}
 
-    std::ofstream out("/tmp/out.csv");
+// Dump the angle trajectory to a file. For debugging purposes.
+[[maybe_unused]]
+static void dump_trajectory(const std::string& file_name, const Trajectory& trajectory, const AngleFunction& angle_fn) {
+    std::ofstream out(file_name);
     auto cursor = trajectory.cursor(trajectory.begin_time());
     for (; cursor.has_value(); cursor.advance(0.01s)) {
         RobotInstant instant = cursor.value();
+
+        double target = angle_fn(instant.linear_motion(), instant.heading(), nullptr);
+
         out << RJ::Seconds(instant.stamp - trajectory.begin_time()).count()
-            << ", " << instant.heading() << ", " << instant.angular_velocity()
-            << ", "
-            << angle_fn(instant.linear_motion(), instant.heading(), nullptr)
+            << ", " << instant.heading()
+            << ", " << instant.angular_velocity()
+            << ", " << target
             << std::endl;
     }
 }
+
+TEST(AnglePlanning, obeys_angle_function_target) {
+    auto trajectory = make_straight_line_trajectory();
+    auto constraints = make_rotation_constraints();
+
+    auto angle_fn = AngleFns::facePoint(Point(2, 0.5));
+
+    // Create a robot instant. This should allow us to have perfect tracking, because we will start perfectly in line with the angle function.
+    RobotInstant initial_instant = make_initial_instant(trajectory.first().linear_motion(), trajectory.first().stamp, angle_fn);
+
+    PlanAngles(&trajectory, initial_instant, angle_fn, constraints);
+
+    check_angle_planning_near_exact(trajectory, angle_fn, 5e-2);
+}
+
+TEST(AnglePlanning, DISABLED_start_from_incorrect_angle) {
```

Comment:
Whats the thought behind this being disabled?

Reply:
It is a desired feature of angle planning that is currently not implemented.